app/src/main/java/com/github/polypoly/app/WelcomeActivity.kt

```diff
@@ -17,21 +17,41 @@ import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import com.github.polypoly.app.global.GlobalInstances.Companion.isSignedIn
+import com.github.polypoly.app.game.GameLobby
+import com.github.polypoly.app.game.GameMode
+import com.github.polypoly.app.game.user.Skin
+import com.github.polypoly.app.game.user.Stats
+import com.github.polypoly.app.game.user.User
+import com.github.polypoly.app.global.GlobalInstances
+import com.github.polypoly.app.global.GlobalInstances.Companion.remoteDB
+import com.github.polypoly.app.global.Settings
+import com.github.polypoly.app.global.Settings.Companion.DB_GAME_LOBIES_PATH
+import com.github.polypoly.app.global.Settings.Companion.DB_USERS_PROFILES_PATH
 import com.github.polypoly.app.menu.JoinGameLobbyActivity
 import com.github.polypoly.app.menu.MenuComposable
 import com.github.polypoly.app.menu.kotlin.GameMusic
+import com.github.polypoly.app.network.RemoteDB
 import com.github.polypoly.app.ui.theme.PolypolyTheme
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
 
 /**
  * This activity is the view that a player will see when launching the app, the idea is that
  * this screen represents the "hub" from where all the main actions are made.
  *
  * These actions may be: creating a game, joining a game, logging in, settings, rules, leaderboards etc.
  */
-class WelcomeActivity : ComponentActivity() {
+class WelcomeActivity : ComponentActivity(){
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent { WelcomeContent() }
+        if (!isSignedIn) {
+            finish()
```

Comment:
You might be right, I think this might be an **artifact** of when I used to implement this the other way around (**WelcomeActivity** was the main, and **Signin** was called when needed). I will check it in detailed, in case there was a real reason why I left it here.

Also for further explanation on why **SigninActivity** finishes,  **SigninActivity** in fact always finishes when passing to **WelcomeActivity**. However, the only case where **WelcomeActivity** finishes itself is when we sign off the user from **Settings**. What happens in **Settings** is that it signs the user off, launches the **SigninActivity** from intent, then finishes itself, which will cause the **WelcomeActivity** to finish itself too right after that, by readingthe `onResume `function. Then the SigninActivity is launched :)

Reply:
Okay I understand better! If it works then it is perfect, thanks for the explanation :)